src/main/java/command/Command.java

```diff
@@ -0,0 +1,36 @@
+package command;
+
+import henry.TaskList;
+
+/**
+ * The command class is the base for all commands.
+ * It only has one method, execute, which returns a CommandResult.
+ * This method is to be implemented by all subclasses of Command.
+ */
+public class Command {
+
+    protected TaskList taskList;
+
+    /**
+     * Returns a CommandResult object based on the type of Command
+     * executed.
+     *
+     * @return
+     */
+    public CommandResult execute() {
+        throw new UnsupportedOperationException("Cannot execute abstract command!");
```

Comment:
what was the reasoning behind throwing an exception instead of making the class abstract?

Reply:
No reason against making the class abstract, I'll add it in a future commit! Thanks!